js/components/negative-controls.js

```diff
@@ -425,10 +423,13 @@ define(['knockout',
 
 			// Kick the job off
 			$.when(negativeControlsJob)
-				.done(function (jobInfo) {
-					pollTimeout = setTimeout(function () {
-						sharedState.jobListing.queue(job);
-						self.pollForInfo();
+				.done(function (info) {
+                    job.name = info.data.jobParameters.jobName;
+                    job.executionId = info.data.executionId;
+                    job.status(info.data.status);
+                    sharedState.jobListing.queue(job);
```

Comment:
1. The 4 lines of code above look exactly the same way as they do in `negative-controls.js`. Why not to move it to util method?
1. Why `queue` here is called outside of timeout function (which seems to be the proper way), while above - inside the timeout?